src/components/Nav.svelte

```diff
@@ -1,17 +1,20 @@
 <script>
   import { state } from "../auth.js";
   import { onMount } from "svelte";
-  var value = [];
+  var value;
   var name;
   var redirectUrl;
   export let segment;
 
   onMount(async () => {
     redirectUrl = window.location.origin;
     state.subscribe(checkstate => {
-      value.push(checkstate);
-
-      name = value[2];
+      if ($state.status === "SUCCESS") {
+        name = $state.name;
+        value = $state.token;
+      } else {
+        status = "FAILED";
+      }
     });
   });
```

Comment:
I'm not sure you need to call `state.subscribe` at all. If I understand correctly, you can just refer to `$state` (or, with the above name change, `$authState`) in the template code below.